pkg/controller/launch/pool_manager.go

```diff
@@ -113,13 +135,31 @@ func (cpm *ConcretePoolManager) SetStatus(status *PoolStatus) error {
 		return err
 	}
 
-	for i, curInfo := range copy.Status.NodePools {
-		if curInfo.Name == newInfo.Name {
-			if curInfo == newInfo {
+	// Add new pools in the spec to the status
+	for _, specPool := range copy.Spec.NodePools {
+		// Find the pool
+		if npi, ok := nodePoolInfoGet(copy.Status.NodePools, specPool.Name); ok {
+			// is there a need to update?
+			if copy.Status.NodePools[npi] == newInfo {
 				return nil
 			}
+			copy.Status.NodePools[npi] = newInfo
+			_, err = cpm.Clients.Kubernikus.Kubernikus().Klusters(copy.Namespace).Update(copy)
```

Comment:
Triggering an update for every change pool seems excessive. I would just set update an `updated` Boolean flag to true and to one update after both ranges.